### PR TITLE
Fix `SafeArea` DartPad sample

### LINKED ISSUE #159340: SafeArea code sample not rendering properly
There turns out to be a bit of a problem with https://github.com/flutter/flutter/pull/158019:

<p align="center">
  RenderFlex overflowed
</p>

I forgot to take the [interactive sample](https://main-api.flutter.dev/flutter/widgets/SafeArea-class.html#widgets.SafeArea.1)'s small screen size into account.

### PR DESCRIPTION
**Pull request by nate-thegrate**
This pull request fixes a bug I introduced in https://github.com/flutter/flutter/pull/158019.

<br>

<h3 align="center">Before</h3>
<p align="center">
  RenderFlex overflowed
</p>

<h3 align="center">After</h3>
<p align="center">
  RenderFlex overflowed
</p>

<br><br>

- fixes https://github.com/flutter/flutter/issues/159340

### CODE REVIEW

File: examples/api/lib/widgets/safe_area/safe_area.0.dart [hunk 1]
```diff
@@ -122,6 +122,7 @@ enum Inset implements Value {
     Text(label),
     Builder(
       builder: (BuildContext context) => Slider(
+        key: ValueKey<Inset>(this),
```
Comment (chunhtai) [inline examples/api/lib/widgets/safe_area/safe_area.0.dart:125]: why is this needed?

Comment (nate-thegrate) [reply examples/api/lib/widgets/safe_area/safe_area.0.dart:125]: I did it to make the testing easier… now that you mention it, I could probably just loop through every slider that the finder returns.

Comment (nate-thegrate) [reply examples/api/lib/widgets/safe_area/safe_area.0.dart]: Factored this out in https://github.com/flutter/flutter/pull/159344/commits/dfef025ba68c7daff5c0eb3e81855dac8f6fa419!

**Review by chunhtai — APPROVED** [review]
LGTM
